src/geom/SphPoint.cc

```diff
@@ -0,0 +1,244 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program. If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <string>
+#include <sstream>
+
+#include <cmath>
+
+#include "Eigen/Geometry"
+
+#include "lsst/afw/geom/SphPoint.h"
+#include "lsst/pex/exceptions.h"
+
+namespace pexExcept = lsst::pex::exceptions;
+using namespace std;
+
+namespace lsst {
+namespace afw {
+namespace geom {
+    /**
+     * Wraps an angle to the interval [0, 2&pi;).
+     *
+     * @param angle the angle to be wrapped
+     * @return an angle equivalent to @c angle, but forced into the
+     *         interval [0, 2&pi;). May be the same object as @c angle if
+     *         no wrapping was required.
+     */
+    Angle wrap(Angle const & angle) {
+        Angle copy(angle);
+        copy.wrap();
+        return copy;
+    }
+
+    /**
+     * Angular distance between two points using the Haversine formula.
+     *
+     * Besides the differences between the two coordinates, we also input the
+     * cosine of the two declinations, so as to be thrifty with the use of
+     * trigonometric functions.
+     *
+     * @param deltaLon Difference between longitudes.
+     * @param deltaLat Difference between latitudes. Must be in the same
+     *                 sense as @c deltaLon.
+     * @param cosLat1, cosLat2 Cosines of the two points' latitudes.
+     * @return the distance between the two points
+     */
+    Angle haversine(Angle const & deltaLon,
+                    Angle const & deltaLat,
+                    double cosLat1,
+                    double cosLat2
+    ) {
+        double const sinLat = sin(deltaLat/2.0);
+        double const sinLon = sin(deltaLon/2.0);
+        double const havDDelta = sinLat * sinLat;
+        double const havDAlpha = sinLon * sinLon;
+        double const havD = havDDelta + cosLat1 * cosLat2 * havDAlpha;
+        double const sinDHalf = sqrt(havD);
+        return (2.0 * asin(sinDHalf)) * radians;
+    }
+
+    SphPoint::SphPoint(Angle const & longitude, Angle const & latitude) :
+                    _longitude(wrap(longitude).asRadians()),
+                    _latitude(latitude.asRadians()) {
+        if (!isfinite(_latitude) || !isfinite(_longitude)) {
+            throw pexExcept::InvalidParameterError("NaN detected in constructing (" +
+                                                   to_string(longitude.asDegrees()) +
+                                                   ", " + to_string(latitude.asDegrees()) +
+                                                   ")");
+        }
+        // For some reason abs(double) is ignored
```

Comment:
http://stackoverflow.com/a/13792680 explains. You must use `std::abs` not `abs`. You can also use `fabs` or `std::fabs` if you prefer.

I personally suggest you always make `std::` explicit. I realize it is a matter of taste, and you are free to do as you prefer, but I personally find it helps clarify where symbols come from (a bit), and it avoids this very issue you ran into.

Please look through the rest of the code for `abs` and change it. I found at least two other instances (which I will note, but I may not find them all).